internal/backend/triggers/triggers.go

```diff
@@ -53,3 +97,98 @@ func (m *triggers) Get(ctx context.Context, triggerID sdktypes.TriggerID) (sdkty
 func (m *triggers) List(ctx context.Context, filter sdkservices.ListTriggersFilter) ([]sdktypes.Trigger, error) {
 	return m.db.ListTriggers(ctx, filter)
 }
+
+func (m *triggers) createScheduledWorkflow(ctx context.Context, trigger sdktypes.Trigger, schedule sdktypes.Value) (sdktypes.Trigger, error) {
+	scheduleStr, err := schedule.ToString()
+	if err != nil || scheduleStr == "" {
+		return trigger, fmt.Errorf("failed to parse trigger schedule <%v>: %w", schedule, err)
+	}
+
+	event := kittehs.Must1(sdktypes.EventFromProto(&sdktypes.EventPB{EventType: "scheduler"}))
+	eventID, err := m.events.Save(ctx, event)
+	if err != nil {
+		return trigger, fmt.Errorf("create scheduler workflow: save event: %w", err)
+	}
+
+	z := m.z.With(zap.String("event_id", eventID.String())).With(zap.String("schedule", scheduleStr))
+	z.Debug("create scheduled event", zap.String("schedule", scheduleStr))
+
+	scheduleID := fmt.Sprintf("%s_%s", trigger.Name().String(), trigger.ID().String())
+	triggerID := trigger.ID()
+	_, err = m.tmprl.ScheduleClient().Create(ctx,
+		client.ScheduleOptions{
+			ID: scheduleID,
+			Spec: client.ScheduleSpec{
+				CronExpressions: []string{scheduleStr},
+			},
+			Action: &client.ScheduleWorkflowAction{
+				ID:        eventID.String(), // workflowID
+				Workflow:  "events_workflow",
+				TaskQueue: "events-task-queue",
+				Args: []interface{}{
+					dispatcher.EventsWorkflowInput{EventID: eventID, TriggerID: &triggerID},
+				},
+			},
+		})
+	if err != nil {
+		m.z.Error("Failed starting scheduler workflow, orphaned event", zap.Error(err))
+		return trigger, fmt.Errorf("failed to create scheduler workflow: %w", err)
+	}
+	z.Debug("created scheduler workflow", zap.String("schedule_id", scheduleID))
+
+	// update trigger with scheduleID
+	trigger = trigger.WithUpdatedData("schedule_id", sdktypes.NewStringValue(scheduleID))
+	return trigger, nil
+}
+
+func extractScheduleID(data map[string]sdktypes.Value) *string {
+	scheduleIDVal, found := data["schedule_id"]
+	scheduleID, err := scheduleIDVal.ToString()
+
+	if !found || err != nil || scheduleID == "" {
+		return nil
+	}
+	return &scheduleID
+}
+
+func (m *triggers) deleteSchedulerWorkflow(ctx context.Context, data map[string]sdktypes.Value) error {
+	scheduleID := extractScheduleID(data)
+	if scheduleID == nil {
+		m.z.Error("Failed delete scheduler workflow. No scheduleID found")
+		return fmt.Errorf("delete scheduler workflow: no scheduleID")
+	}
+
+	scheduleHandle := m.tmprl.ScheduleClient().GetHandle(ctx, *scheduleID) // validity of scheduleID is not checked by temporal
+	if err := scheduleHandle.Delete(ctx); err != nil {
+		return fmt.Errorf("delete scheduler workflow: %w", err)
+	}
+	// REVIEW: create and save cancel schedule event?
```

Comment:
Is this a task before merging, or does this need a ticket?

Reply:
one more open question. @itayd @daabr WDYT?